### PR TITLE
Added r-rang as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
Needs https://github.com/conda-forge/staged-recipes/pull/23639

**Comment by thewchan** [comment]
@conda-forge/staged-recipes @conda-forge/help-r there's some win build dependencies issue but this is no arch anyway. Otherwise it's ready for review!

File: recipes/r-rang/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    - r-vctrs
+
+test:
+  commands:
+    - $R -e "library('rang')"           # [not win]
+    - "\"%R%\" -e \"library('rang')\""  # [win]
+
+about:
+  home: https://github.com/chainsawriot/rang
+  license: GPL-3.0-only
```
Comment (mfansler) [inline recipes/r-rang/meta.yaml:59]: CRAN shows `GPL (>=3)`, so...

```suggestion
  license: GPL-3.0-or-later
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for another one! Minor touch-up on the license; otherwise 👌

**Pull request merged**